File: app/src/main/java/com/example/jetpackcomponentsapp/view/BaseViewPagerHolder.java
package com.example.jetpackcomponentsapp.view;

import android.content.Context;
import android.util.Log;
import android.view.View;

import com.example.jetpackcomponentsapp.model.CustomModel;

abstract public class BaseViewPagerHolder extends BaseViewHolder {

    static private final String TAG = BaseViewPagerHolder.class.getSimpleName();

    public BaseViewPagerHolder(Context context, View itemView) {
        super(context, itemView);
        Log.d(TAG, "constructor");
    }

    abstract public void bindDataToViewHolder(CustomModel customModel, int position);
}

File: app/src/main/java/com/example/jetpackcomponentsapp/view/BaseViewPagerHolder.kt
package com.example.jetpackcomponentsapp.view

import android.util.Log
import android.view.View
import com.example.jetpackcomponentsapp.model.CustomModel

abstract class BaseViewPagerHolder : BaseViewHolder {

    companion object {
        private val TAG = BaseViewPagerHolder::class.java.getSimpleName()
    }

    constructor(itemView : View) : super(itemView) {
        Log.d(TAG, "constructor")
    }

    abstract fun bindDataToViewHolder(customModel : CustomModel, position : Int)
}